### ISSUE TITLE
Cannot access App Services in VS code (Internal Error)

### ISSUE THREAD

**Issue by wittysuyu** [microsoft/vscode-azureappservice#2661]
Issue
====
Not able to load azure resources in VS code using azure CLI... 

Operation 'Microsoft.Web/sites/read' failed as server encountered too many requests. Please try after '60' seconds. Tracking Id is '663dbaad-a145-4031-bf3e-04b2c45dba9c'.

Go to visual studio code
Sign into Azure
Visit the App Services under a specific subscription
Cannot load them

unable to load app services in sub  844f89f3-af43-43a6-9519-37b8e95ca138 only
![image](https://github.com/user-attachments/assets/04dcab65-bce2-4031-a357-aa8b7aee8f03)
the other subs are all good
![image](https://github.com/user-attachments/assets/6eb3a915-a756-4f04-bb71-19108e5beda9)
tried to use command line to check directly and is able to reach the resources and deploy
![image](https://github.com/user-attachments/assets/310efc2a-3c5f-450b-9657-1199ef9a8f38)
from Azure portal, have the access to view the exact resource as well
![image](https://github.com/user-attachments/assets/27cdbef4-08f6-4341-bdf9-598902815c97)
only have issue list the resource in VS code portal
![image](https://github.com/user-attachments/assets/09101eba-1e14-42f3-9f8a-306a318117be)
![image](https://github.com/user-attachments/assets/a3426eb4-9fa3-414f-b748-3c58581e4c59)

found a similar issue raised recently and is with logic app https://github.com/Azure/LogicAppsUX/issues/4929, please help check and advise

**Comment by nturinski** [comment]
Reopening until we confirm that the fix actually works and is released.

**Comment by nturinski** [comment]
If you could help us test that our fix actually worked, please follow these steps:
1. Download the zip file artifact from here: https://github.com/microsoft/vscode-azureappservice/actions/runs/10516435613/artifacts/1844660618
2. Extract the `.vsix` file from the zip file
3. In VS Code, hit F1 > `Extensions: Install from VSIX`
4. Navigate to `vscode-azureappservice.vsix`
5. Install and reload VS Code

If this resolves your issue, please let us know. It would help us greatly!

**Comment by wittysuyu** [comment]
Thanks @nturinski for the suggestions. I will check with users having this issue and verify if it could work. Will post here once have any updates.

**Comment by nturinski** [comment]
This fix is released. Please let me know if the users have had more success with the new release!

**Comment by nturinski** [comment]
We don't have an official RCA or anywhere to to file it, but I can explain here. 

The issue was caused by another extension loading all of the resources in parallel which caused the old caching logic to not work which resulted in many more requests to the Azure API than we had intended.

**Comment by wittysuyu** [comment]
Thanks @nturinski for the explanation! Just per some further feedback from my customers that the issue has been resolved for him. However, some other colleagues are still encountering the same problem occasionally after following the steps outlined. Additionally, loading app names in VS Code is taking longer than expected.

If you have any suggestions or further steps to address these issues, could you please share them with us?

**Comment by nturinski** [comment]
Hi @wittysuyu

Could you please share what versions of the Resources extension and the Azure App Service extension that they have installed?

**Comment by wittysuyu** [comment]
Thanks @nturinski for the additional attention. As I have confirmed with my customer, since those who still getting the issue intermittently are not my direct cx, it is hard to check info with them. and since the issue is not so frequent, we agreed to put the issue on hold and will inform them to upgrade the extension to the latest just in case of next time they have similar issue. Thank you for your patience and explanation so far.
